inject.js

```diff
@@ -1,65 +1,71 @@
 const nest = require('depnest');
-const index = require('./index');
-const {
-  h
-} = require('mutant');
+const { h, onceTrue } = require('mutant');
+// const { isMsg } = require('ssb-ref')
 
-var onceTrue = require('mutant/once-true');
+const index = require('./index');
 
 exports.gives = nest({
-  'app.html': {
-    page: true,
-    menuItem: true
-  }
+  'app.html.menuItem': true, 
+  'app.page.chessIndex': true,
+  'router.sync.routes': true,
 })
 
 exports.needs = nest({
+  'app.page.chessIndex': 'first',
   'sbot.obs.connection': 'first',
   'backlinks.obs.for': 'first'
 });
 
 exports.create = function(api) {
-  const route = '/chess'
-
-  const topLevelDomElement = document.createElement('div');
-  topLevelDomElement.className = "ssb-chess-container";
-
+  const topLevelDomElement = h('div.ssb-chess-container', { title: '/chess' })
   var pageLoaded = false;
 
   return nest({
-    'app.html': {
-      menuItem: menuItem,
-      page: chessPage
-    }
+    'app.html.menuItem': menuItem, 
+    'app.page.chessIndex': chessIndex,
+    'router.sync.routes': routes,
   })
 
   function menuItem(handleClick) {
     return h('a', {
       style: {
         order: 0
       },
-      'ev-click': () => handleClick(route)
-    }, route)
+      'ev-click': () => handleClick({ page: 'chess' })
+    }, '/chess')
   }
 
-  function chessPage(path) {
-    if (path !== route) {
-      return;
-    } else if (pageLoaded) {
+  function chessIndex(location) {
```

Comment:
The router is responsible for whether something gets here so we don't need to check if the location (formally path)  matches.

Given you've got a return on line 40, I've taken the liberty of dropping the else block (as it's not needed)